chain/chain/src/chain.rs

```diff
@@ -2063,7 +2070,11 @@ impl Chain {
         let shard_state_header = self.get_state_header(shard_id, sync_hash)?;
```

Comment:
This function `set_state_part` can also be changed to use PartId

Reply:
There might be other functions in the code that are using `part_id` + `num_parts` that can be replaced with `PartId`, I didn't check very carefully that these are the only three